Hazel/Udp/UnityUdpClientConnection.cs

```diff
@@ -49,14 +50,26 @@ protected override void WriteBytesToConnection(byte[] bytes, int length)
         {
             try
             {
-                socket.BeginSendTo(
-                    bytes,
-                    0,
-                    length,
-                    SocketFlags.None,
-                    EndPoint,
-                    HandleSendTo,
-                    null);
+                if (this.sendSynchronously)
```

Comment:
I have a tenuous relationship with Disconnect(), since I don't know if/why anyone would want to disconnect without disposing, so since I expect that Disconnect is only called when the connecting is about to die anyway, I prefer that the SendTo is isolated to SendDisconnect(), this would mirror the implementations in the other UdpConnection classes.

If the location is not okay, I'm also okay enforcing a protected abstract WriteBytesToConnectionSync method in UdpConnection, but the net result should be that same. I just fear a client sending on the main thread just as this dispose fires, so the main thread goes sync and potentially causes issues.